tfexec/internal/testutil/tfcache.go

```diff
@@ -79,7 +73,10 @@ func (tf *TFCache) find(t *testing.T, key string, finder func(dir string) tfinst
 		panic(fmt.Sprintf("unable to mkdir %q: %s", dir, err))
 	}
 
-	path, err := tfinstall.Find(context.Background(), finder(dir))
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
```

Comment:
there isn't a reason to use a cleanup here over a defer since its already function scoped, but I'm curious why use a cancelable context at all?

Reply:
defer != Cleanup, defer runs at the end of the function execution, i.e. when `find()` returns, whereas `t.Cleanup()` will run at the end of the test - e.g. after `t.Fatal()` or just when the test passes.

The cancellable context is useful when you want to just "Ctrl+C" running tests which are currently installing Terraform - presumably that would otherwise hang until the installation finishes.

Reply:
I see so `t.Cleanup` runs when you press ctrl+c?

Reply:
This doesn't seem to happen for me in practice in a test project. I don't think cleanups are run when you Ctrl+C, at least on Windows.

Reply:
I can cancel the tests as well without waiting regardless of context.

Reply:
https://golang.org/pkg/os/signal/#hdr-Default_behavior_of_signals_in_Go_programs

SIGINT just causes it to exit unless a Notify has changed things from the default behavior.

Reply:
I assume you're right (haven't tested it myself), but I still think this may be still useful for when the context is cancelled due to the timeout (i.e. exceeded `-timeout` as a flag given to `go test`), which doesn't involve any signals.

I'm going to merge it as is, but feel free to raise a separate issue or PR if you wish to discuss further.